#167190542 Implement Article Tagging

henryade opened the pull request:
#### What does this PR do?
Enables Users Tag Articles

#### Description of Task proposed in this pull request?
- Create tag migration, model and seed files
- Create an article tag migration file
- Add functionality that allows users to tag their articles.
- Write model, function, and end to end test
- Implement get all tag route

#### How should this be manually tested (Quality Assurance)?
- Run `npm run db:rollback` to get all tables
##### For Tag Get Route
- Make a `GET` request to `/api/v1/tags`
##### For Article Tagging
- On the creation of the Article, you will be required to add tags. Those tags will be created with the articles. 
Route `POST` `api/v1/article/create` [Protected]
body: 
{
"title": "Success In Life',
"status": "publish",
"articleBody": "Failure is the best Lecturer for a Success course",
"tags": "tech,motorbike"
}

#### What are the relevant pivotal tracker stories?
- Story ID:  [#167190542](https://www.pivotaltracker.com/story/show/167190542)

#### Any background context you want to add (Operations Impact)?
- Run `npm run db:rollback` to get all tables

#### What I have learned working on this feature: 
- I learned more about sequelize magic methods

#### Screenshots: 
Get all Articles:
<img width="1440" alt="Screenshot 2019-08-15 at 7 01 51 AM" src="https://user-images.githubusercontent.com/47576037/63075755-af2cef00-bf2a-11e9-87ab-5760b8ce0f6b.png">

Tag Article:
<img width="1440" alt="Screenshot 2019-08-16 at 12 35 29 PM" src="https://user-images.githubusercontent.com/47576037/63165174-8ee05b80-c022-11e9-9a79-0956f4b9d20b.png">

Proposed Mockup:
<img width="1440" alt="Screenshot 2019-08-16 at 9 36 35 AM" src="https://user-images.githubusercontent.com/47576037/63155337-41f08b00-c00a-11e9-8768-37356daa1e2c.png">

chialuka commented on test/tags/createTag.test.js:
```diff
@@ -0,0 +1,45 @@
+    });
+  });
+
+  context('when an array of existing tags is passed in', () => {
```
This context block and the one following are the same

chialuka reviewed (commented):
Good job

henryade committed:
feature-[167190542]: implement article tagging
- create tag migration, model and seed files
- create articletag migration file
- enable user tag article
- write model, helper and end to end test
- enable admin create category
- implement get all tag route
[Delivers #167190542]

JuwonAbiola reviewed (approved):
Works fine
